iocore/eventsystem/test_Event.cc

```diff
@@ -91,23 +91,13 @@ reconfigure_diags()
 static void
 init_diags(const char *bdt, const char *bat)
 {
-  FILE *diags_log_fp;
   char diags_logpath[500];
   strcpy(diags_logpath, DIAGS_LOG_FILE);
 
-  diags_log_fp = fopen(diags_logpath, "w");
-  if (diags_log_fp) {
-    int status;
-    status = setvbuf(diags_log_fp, NULL, _IOLBF, 512);
-    if (status != 0) {
-      fclose(diags_log_fp);
-      diags_log_fp = NULL;
-    }
-  }
-
-  diags = new Diags(bdt, bat, diags_log_fp);
+  BaseLogFile *blf = new BaseLogFile(diags_logpath);
+  diags = new Diags(bdt, bat, blf);
 
-  if (diags_log_fp == NULL) {
+  if (blf && blf->m_fp == NULL) {
```

Comment:
How can `blf` be `NULL` since you just allocated it? And in that case the condition should be `NULL == blf || NULL == blf->m_fp` because if `blf` is NULL you're failed as well.